screeps-game-api/src/constants.rs

```diff
@@ -59,54 +61,91 @@ impl ReturnCode {
 
 js_deserializable!(ReturnCode);
 
-pub unsafe trait FindConstant {
-    type Item: FromExpectedType<Reference>;
+/// Constants for use with the [`Room::find`] function.
+///
+/// *Note:* Types in this module have purposefully ambiguous names, and are
+/// intended to be used as, for example, `find::CREEPS`, not `CREEPS`.
+///
+/// You can do this my importing the module itself, rather than any indvidiual
+/// constant, and then just referring to the constants relative to the module.
+///
+/// # Example
+///
+/// ```no_run
+/// use screeps::{find, game, Room};
+///
+/// let room: Room = game::rooms().get("E23S55").unwrap();
+///
+/// let creeps = room.find(find::CREEPS);
+/// # let _ = creeps;
+/// ```
+///
+/// [`Room::find`]: crate::Room::find
+/// [`objects::RoomObject`]: crate::RoomObject
+pub mod find {
+    use num_derive::FromPrimitive;
+    use serde_repr::{Deserialize_repr, Serialize_repr};
+    use stdweb::Reference;
 
-    fn find_code(&self) -> i32;
-}
+    use crate::{
+        objects::{
+            ConstructionSite, Creep, Flag, Mineral, Nuke, OwnedStructure, PowerCreep, Resource,
+            RoomPosition, Source, Structure, StructureSpawn, Tombstone,
+        },
+        traits::FromExpectedType,
+    };
 
-#[derive(
-    Debug, PartialEq, Eq, Clone, Copy, Hash, FromPrimitive, Deserialize_repr, Serialize_repr,
-)]
-#[repr(i32)]
-pub enum FindObject {
-    Creeps = 101,
-    MyCreeps = 102,
-    HostileCreeps = 103,
-    SourcesActive = 104,
-    Sources = 105,
-    DroppedResources = 106,
-    Structures = 107,
-    MyStructures = 108,
-    HostileStructures = 109,
-    Flags = 110,
-    ConstructionSites = 111,
-    MySpawns = 112,
-    HostileSpawns = 113,
-    MyConstructionSites = 114,
-    HostileConstructionSites = 115,
-    Minerals = 116,
-    Nukes = 117,
-    Tombstones = 118,
-}
+    /// Trait representing things which can be used in the 'find' function.
+    ///
+    /// Typically used with zero-sized structs in the
+    /// [`find`][crate::constants::find] module.
+    pub unsafe trait FindConstant {
+        type Item: FromExpectedType<Reference>;
 
-unsafe impl FindConstant for FindObject {
-    type Item = RoomObject;
+        fn find_code(&self) -> i32;
+    }
 
-    fn find_code(&self) -> i32 {
-        *self as i32
+    /// Useful for finding any [`RoomObject`][crate::objects::RoomObject] with
+    /// a dynamically-chosen find constant.
+    ///
+    /// If you know ahead of time what constant you'll use, then the
+    /// all-upper-case constants in [this module](crate::constants::find) will
+    /// be more helpful, and won't require casting the result types.
+    ///
+    /// If you do use this constant, I recommend just importing the `find`
```

Comment:
That sounds better, yeah. I wanted to write something "softer" than that, but since code really does look weird without `find::` beforehand "should" does seem good.

Reply:
In other places I seem to use "it's recommended", which is slightly better. But _should_ in this case seems right, still.

Reply:
_You may want_ ?